### PR TITLE
Add CookieManager implementation

### PR DESCRIPTION
**Pull request by Syer10**
Fixes sources like MangaFox and Tapas, allowing them to insert thier own cookies into the cookie store.

### CODE REVIEW

File: server/src/main/kotlin/eu/kanade/tachiyomi/network/NetworkHelper.kt [hunk 1]
```diff
@@ -32,12 +35,19 @@ class NetworkHelper(context: Context) {
 
 //    private val cacheSize = 5L * 1024 * 1024 // 5 MiB
 
-    val cookieManager = PersistentCookieJar(context)
+    // Tachidesk -->
+    val cookies = PersistentCookieStore(context)
```
Comment (schroda) [inline server/src/main/kotlin/eu/kanade/tachiyomi/network/NetworkHelper.kt]: I'd name it `cookieStore`

File: server/src/main/kotlin/eu/kanade/tachiyomi/network/PersistentCookieStore.kt [hunk 2]
```diff
@@ -32,48 +40,153 @@ class PersistentCookieStore(context: Context) {
 
     @Synchronized
```
Comment (schroda) [inline server/src/main/kotlin/eu/kanade/tachiyomi/network/PersistentCookieStore.kt]: still needed with ReentrantLock? (same for `removeAll`)

File: server/src/main/kotlin/eu/kanade/tachiyomi/network/NetworkHelper.kt (hunk 1 above)
Comment (Syer10) [reply server/src/main/kotlin/eu/kanade/tachiyomi/network/NetworkHelper.kt]: Done 👍

File: server/src/main/kotlin/eu/kanade/tachiyomi/network/PersistentCookieStore.kt (hunk 2 above)
Comment (Syer10) [reply server/src/main/kotlin/eu/kanade/tachiyomi/network/PersistentCookieStore.kt]: Done 👍

**Review by schroda — APPROVED** [review]

**Pull request merged**